sdk/tables/Azure.Data.Tables/src/DynamicTableEntity.cs

```diff
@@ -0,0 +1,221 @@
+            set { _properties[TableConstants.PropertyNames.PartitionKey] = value; }
+        }
+
+        /// <summary>
+        /// The row key is a unique identifier for an entity within a given partition. Together the <see cref="PartitionKey" /> and RowKey uniquely identify every entity within a table.
+        /// </summary>
+        /// <value>A string containing the row key for the entity.</value>
+        public string RowKey
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.RowKey, out object value) ? (string)value : default; }
+            set { _properties[TableConstants.PropertyNames.RowKey] = value; }
+        }
+
+        /// <summary>
+        /// The Timestamp property is a DateTime value that is maintained on the server side to record the time an entity was last modified.
+        /// The Table service uses the Timestamp property internally to provide optimistic concurrency. The value of Timestamp is a monotonically increasing value,
+        /// meaning that each time the entity is modified, the value of Timestamp increases for that entity. This property should not be set on insert or update operations (the value will be ignored).
+        /// </summary>
+        /// <value>A <see cref="DateTimeOffset"/> containing the timestamp of the entity.</value>
+        public DateTimeOffset Timestamp
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.TimeStamp, out object value) ? (DateTimeOffset)value : default; }
+        }
+
+        /// <summary>
+        /// Gets or sets the entity's ETag. Set this value to '*' in order to force an overwrite to an entity as part of an update operation.
+        /// </summary>
+        /// <value>A string containing the ETag value for the entity.</value>
+        public string ETag
+        {
+            get { return _properties.TryGetValue(TableConstants.PropertyNames.Etag, out object value) ? (string)value : default; }
+            set { _properties[TableConstants.PropertyNames.Etag] = value; }
+        }
+
+        /// <summary>
+        /// Constructs an instance of a <see cref="TableEntity" />.
+        /// </summary>
+        public DynamicTableEntity()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DynamicTableEntity"/> class with the specified partition key and row key.
+        /// </summary>
+        /// <param name="partitionKey">A string containing the partition key of the <see cref="DynamicTableEntity"/> to be initialized.</param>
+        /// <param name="rowKey">A string containing the row key of the <see cref="DynamicTableEntity"/> to be initialized.</param>
+        public DynamicTableEntity(string partitionKey, string rowKey)
+        {
+            PartitionKey = partitionKey;
+            RowKey = rowKey;
+        }
```

Comment:
You need to initialize `_properties` in both of these .ctors - especially before setting `PartitionKey`.  It should be fine to use `public DynamicTableEntity() : this(null) { ... }` and `public DynamicTableEntity(string partitionKey, string rowKey) : this(null) { ... }`.